docs/explanations/default-options.rst

```diff
@@ -0,0 +1,64 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-default-options:
+
+*********************
+Default build options
+*********************
+
+Meson offers many `built-in options <https://mesonbuild.com/Builtin-options.html>`__,
+and in the vast majority of cases those have good defaults. There are a couple
+of cases however where ``meson-python`` either needs to or chooses to override
+those with its own defaults. To view what those are for the version of
+``meson-python`` you have installed, look at the *User defined options* section
+of the output during the configure stage of the build (e.g., by running
+``python -m build --wheel``). This will look something like:
+
+.. code-block:: text
+
+    User defined options
+      Native files     : /home/username/code/project/.mesonpy-native-file.ini
+      debug            : false
+      optimization     : 2
+      prefix           : /home/username/mambaforge/envs/project-dev
+      python.platlibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      python.purelibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      b_ndebug         : if-release
```

Comment:
possible improvement:

I'd group the options we need to set, with the ones we choose to override the default. Something like this:

```suggestion
      (we need override)
      prefix           : /home/username/mambaforge/envs/project-dev
      python.platlibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
      python.purelibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
      Native files     : /home/username/code/project/.mesonpy-native-file.ini
      (we chose to override)
      debug            : false
      optimization     : 2
      b_ndebug         : if-release
```

It might make sense to make the distinction, so that users know what they can and can't override.

Reply:
This is the output of the `meson setup` command. Presenting an edited version of it that does not match what the uses sees would be confusing.

Reply:
Agreed, I just Ctrl-C'd this from actual build output (replacing my own user name with `username` and package name with `project`) and would prefer to leave it as is.

Reply:
rejecting the suggested change - resolving.